adapter/test/pubsub/pubsub_connection_test.cc

```diff
@@ -0,0 +1,29 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Distributed under BSD 3-Clause license.                                   *
+ * Copyright by The HDF Group.                                               *
+ * Copyright by the Illinois Institute of Technology.                        *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of Hermes. The full Hermes copyright notice, including  *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the top directory. If you do not  *
+ * have access to the file, you may request a copy from help@hdfgroup.org.   *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include <pubsub/pubsub.h>
+
+int main(int argc, char **argv) {
+  hermes::pubsub::mpiInit(argc, argv);
+
+  char *config_file = 0;
+  if (argc == 2) {
+    config_file = argv[1];
+  } else {
+    config_file = getenv(kHermesConf);
+  }
+
+  auto connect_ret = hermes::pubsub::connect(config_file);
```

Comment:
This test is redundant because all your other tests call `connect` and `disconnect`. Please remove it.